rfcs/20230606-topk.md

```diff
@@ -0,0 +1,88 @@
+# RFC: Add TopK Op to StableHLO
+
+Status: Draft<br/>
+Initial version: 2023-06-06<br/>
+Last updated: 2023-06-06
+
+## Summary
+
+We propose to add a new op: `stablehlo.topk`, which returns the first `k`
```

Comment:
nit: Why topk versus DynamicTopK if it supports dynamic operands?

Reply:
I thought about it some more and I think that you are right and static `k` is the better choice

Reply:
Interestingly, we have [a feature request](https://github.com/openxla/stablehlo/issues/1396) from @gnecula about supporting dynamic `k` :)

Reply:
Hmm thanks for that spec! If this op should support dynamic k, what's the StableHLO convention for dynamic ops? For example, Slice has SliceOp and DynamicOp for the static / dynamic pair. If this op should support dynamic k, is this really a DynamicTopK and we'll have a corresponding TopK for static?

Reply:
I'm hoping to discuss this in the upcoming dynamism RFC in more detail (ETA: end of week), but I would propose that we support both in the same TopKOp. Overall, we have received feedback that the Foo/DynamicFoo split makes things harder than they should be for both producers and consumers.